Source/SwiftyDropbox/Shared/Handwritten/Request+TokenRefresh.swift

```diff
@@ -103,26 +125,40 @@ class RequestWithTokenRefresh: ApiRequest {
 
     private func setRequest(_ request: Alamofire.Request) {
         self.request = request
+        setupCompletionHandler()
         setProgressHandlerIfNecessary()
-        setCompletionHandlerIfNecessary()
         if cancelled {
             request.cancel()
         } else {
             request.resume()
         }
     }
 
-    private func setCompletionHandlerIfNecessary() {
-        guard let completionHandler = completionHandler else { return }
-        switch completionHandler {
-        case .dataCompletionHandler(let handler):
-            if let dataRequest = request as? Alamofire.DataRequest {
-                dataRequest.validate().response(queue: responseQueue, completionHandler: handler)
+    private func setupCompletionHandler() {
+        if let dataRequest = request as? Alamofire.DataRequest {
+            let wrappedHandler: (DefaultDataResponse) -> Void = { dataResponse in
+                if case let .dataCompletionHandler(handler) = self.completionHandler {
+                    self.completionHandlerQueue.async {
+                        handler(dataResponse)
+                        self.cleanup()
+                    }
+                } else {
+                    self.cleanup()
```

Comment:
the `else` case here and below seem to me like they should still happen on completionHandlerQueue.async